apps/nifconv/nifconv.cpp

```diff
@@ -0,0 +1,155 @@
+#endif
+
+int main(int argc, char **argv) {
+    namespace bpo = boost::program_options;
+    bpo::options_description desc("Syntax: nifconv <options>\nAllowed options");
+
+    desc.add_options()
+        ("help", "print help message")
+        ("version", "print version information and quit")
+        ("fs-strict", bpo::value<bool>()->implicit_value(true)
+            ->default_value(false), "strict file system handling (no case folding)")
+        ("binary", bpo::value<bool>()->implicit_value(true)
+            ->default_value(false), "write OSG file in binary format (OSGB)")
+        ("data",
+         bpo::value<Files::PathContainer>()->default_value(Files::PathContainer(), "data")
+            ->multitoken()->composing(),
+         "set data directories (later directories have higher priority)")
+        ("data-local", bpo::value<std::string>()->default_value(""),
+         "set local data directory (highest priority)")
+        ("fallback-archive",
+         bpo::value<StringsVector>()->default_value(StringsVector(), "fallback-archive")
+         ->multitoken(), "set fallback BSA archives (later archives have higher priority)")
+        ("nif", bpo::value<StringsVector>(), "NIF file to convert")
+        ;
+
+    bpo::positional_options_description p;
+    p.add("nif", -1);
+
+    // Runtime options override settings from all configs
+    bpo::parsed_options valid_opts = bpo::command_line_parser(argc, argv)
+        .options(desc).positional(p).allow_unregistered().run();
+    bpo::variables_map variables;
+    bpo::store(valid_opts, variables);
+    bpo::notify(variables);
+
+    if (variables.count("help")) {
+        std::cout << desc << std::endl;
+        return false;
+    }
+
+    Files::ConfigurationManager cfgMgr;
+    cfgMgr.readConfiguration(variables, desc);
+
+    bool mFSStrict = variables["fs-strict"].as<bool>();
+    Files::PathContainer dataDirs(variables["data"].as<Files::PathContainer>());
+
+    std::string local(variables["data-local"].as<std::string>());
+    if (!local.empty()) {
+        dataDirs.push_back(Files::PathContainer::value_type(local));
+    }
+
+    cfgMgr.processPaths(dataDirs);
+
+    Files::Collections mFileCollections = Files::Collections (dataDirs, !mFSStrict);
+
+    std::vector<std::string> mArchives;
+    std::auto_ptr<Resource::ResourceSystem> mResourceSystem;
+
+    // fallback archives
+    StringsVector archives = variables["fallback-archive"].as<StringsVector>();
+    BOOST_FOREACH(const std::string a, archives) {
+        mArchives.push_back(a);
+    }
+
+    // engine.setResourceDir(variables["resources"].as<std::string>());
+        
+    std::auto_ptr<VFS::Manager> mVFS;
+    mVFS.reset(new VFS::Manager(mFSStrict));
+    VFS::registerArchives(mVFS.get(), mFileCollections, mArchives, true);
+    mResourceSystem.reset(new Resource::ResourceSystem(mVFS.get()));
+
+    Resource::SceneManager* sceneManager = mResourceSystem->getSceneManager();
+
+#ifdef ACTIVATE
```

Comment:
Agreed, and that's working -- at least enough that osgconv and osgviewer will load the files with the custom extensions with out complaining.   They don't always display correctly, and I assume that's because we've not pulled everything need to act on the objects into the OSG render code?